api/server/router/local/image.go

```diff
@@ -184,12 +177,7 @@ func (s *router) postImagesPush(ctx context.Context, w http.ResponseWriter, r *h
 
 	name := vars["name"]
 	output := ioutils.NewWriteFlusher(w)
-	imagePushConfig := &graph.ImagePushConfig{
-		MetaHeaders: metaHeaders,
-		AuthConfig:  authConfig,
-		Tag:         r.Form.Get("tag"),
-		OutStream:   output,
-	}
+	imagePushConfig := graph.NewImagePushConfig(metaHeaders, authConfig, r.Form.Get("tag"), output)
```

Comment:
> because whole point of structs was to not guess parameters positions.

You still need to know the struct fields to build it. Moving the constructor to a function helps to manage the struct initialization by avoiding repetition. If we have to add/remove/modify a field, you only need to do it in one place, rather than every time you use the struct.